zio-config/src/main/scala/zio/config/actions/ConfigDocs.scala

```diff
@@ -1,49 +1,100 @@
 package zio.config.actions
 
 import zio.config.ConfigDescriptor
-import zio.config.actions.ConfigDocs.KeyDescription
 
-final case class ConfigDocs(configKeysAndDescription: List[KeyDescription], or: Option[ConfigDocs])
+sealed trait ConfigDocs
 
 // Man page
 object ConfigDocs {
-  final case class KeyDescription(path: String, list: List[String])
 
-  final def docs[A](config: ConfigDescriptor[A]): ConfigDocs = {
+  final case class Empty()                                  extends ConfigDocs
+  final case class Leaf(value: KeyDescription)              extends ConfigDocs
+  final case class And(left: ConfigDocs, right: ConfigDocs) extends ConfigDocs
+  final case class Or(left: ConfigDocs, right: ConfigDocs)  extends ConfigDocs
+
+  final case class KeyDescription(path: String, value: Option[String], docs: List[String])
```

Comment:
the `value` is the description?

Reply:
The value is the actual value of the path. User may or may not have given a value of the path. In that case we say the value is None and still provide a description

Reply:
If you could take a look at the example, it will be much clear

Reply:
I see, Thanks :)